src/snowflake/connector/cursor.py

```diff
@@ -1367,13 +1359,13 @@ def reset(self, closing: bool = False) -> None:
         if not self.connection._reuse_results:
             self._result_set = None
 
-    def __iter__(self) -> Iterator[dict] | Iterator[tuple]:
+    def __iter__(self) -> Iterator[dict | None] | Iterator[tuple | None]:
         """Iteration over the result set."""
-        # set _result if _result_set is not None
-        if self._result is None and self._result_set is not None:
-            self._result = iter(self._result_set)
-            self._result_state = ResultState.VALID
-        return self._result_iterator()
+        while True:
+            _next = self.fetchone()
+            if _next is None:
```

Comment:
is it possible that `None` is a valid value in self._result, in such case we exit the loop sooner that some of data are not returned to users